CSVSource/src/test/java/io/vantiq/extsrc/CSVSource/TestCSVReader.java

```diff
@@ -0,0 +1,248 @@
+        assertTrue("Unexpected field field3 in first of line of csv file in array", content.get(0).get("field3").equals("3"));
+
+
+    }
+
+    @Test
+    public void testFredTest() {
+
+
+        CreateFileForTest(testFullFilePath,"fred;namir;marty");
+        AppendFileForTest(testFullFilePath,"s1;11;21");
+
+        
+        config.put("delimiter",";");
+        
+        ArrayList<Map<String,String>> content = CSVReader.execute(testFullFilePath, config, null);
+        assertTrue("Unexpected lines of csv file in array", content.size() == 2);
+        assertTrue("Unexpected values in first of line of csv file in array", content.get(0).size() == 3);
+
+        assertTrue("Unexpected field value in first of line of csv file in array", content.get(0).get("value").equals("fred"));
+        assertTrue("Unexpected field YScale in first of line of csv file in array", content.get(0).get("YScale").equals("namir"));
+        assertTrue("Unexpected field flag in first of line of csv file in array", content.get(0).get("flag").equals("marty"));
+
+
+    }
+
+    @Test
+    public void testFredTest2() {
+
+        CreateFileForTest(testFullFilePath,"fred12345namir33marty");
+        config.put("delimiter","[0-9]");
+        
+        ArrayList<Map<String,String>> content = CSVReader.execute(testFullFilePath, config, null);
+        assertTrue("Unexpected lines of csv file in array", content.size() == 1);
+        assertTrue("Unexpected values in first of line of csv file in array", content.get(0).size() == 3);
+
+        assertTrue("Unexpected field value in first of line of csv file in array", content.get(0).get("value").equals("fred"));
+        assertTrue("Unexpected field YScale in first of line of csv file in array", content.get(0).get("YScale").equals("namir"));
+        assertTrue("Unexpected field flag in first of line of csv file in array", content.get(0).get("flag").equals("marty"));
+
+
+    }
+    @Test
+    public void testSkipNullValues() {
+
+        CreateFileForTest(testFullFilePath,"fred12345namir33marty");
+        config.put("delimiter","[0-9]");
+        
+        ArrayList<Map<String,String>> content = CSVReader.execute(testFullFilePath, config, null);
+        assertTrue("Unexpected lines of csv file in array", content.size() == 1);
+        assertTrue("Unexpected values in first of line of csv file in array", content.get(0).size() == 3);
+
+        assertTrue("Unexpected field value in first of line of csv file in array", content.get(0).get("value").equals("fred"));
+        assertTrue("Unexpected field YScale in first of line of csv file in array", content.get(0).get("YScale").equals("namir"));
+        assertTrue("Unexpected field flag in first of line of csv file in array", content.get(0).get("flag").equals("marty"));
+
+
+    }
+    @Test
+    public void testProccessedNullValues() {
+
+        CreateFileForTest(testFullFilePath,"fred12345namir33marty");
+        config.put("delimiter","[0-9]");
```

Comment:
Since your file has no null values in it, what is this testing?  It's not different than the test above, is it?

Reply:
few days ago i have added new configuration attribute : processNullValues and its value is different between the two tests . 
other then that , the delimiter is [0-9] and not [0-9]* , which mean that when you have 2 adjusted digit , the split function will return null value so fred12345namir33marty will split to fred,,,,,namir,,marty  and now there are two options what to do with the nulls,